packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java

```diff
@@ -80,6 +97,28 @@ ResponseEntity<Model> getModel(
 			return ResponseEntity.noContent().build();
 		}
 
+		// Find the Document Assets linked via provenance to the model
+		ProvenanceQueryParam body = new ProvenanceQueryParam();
+		body.setRootId(id);
+		body.setRootType(ProvenanceType.MODEL);
+		body.setTypes(List.of(ProvenanceType.DOCUMENT));
+		final JsonNode provenanceResults = provenanceProxy.search(body, "models_from_document").getBody();
+		final JsonNode resultsNode = provenanceResults.get("result");
+
+		// If there are results, fetch the Document Assets, gather their extractions
+		if (resultsNode != null && resultsNode.isArray() && !resultsNode.isEmpty()) {
+			final List<String> documentIds = mapper.convertValue(resultsNode, new TypeReference<List<String>>() {});
+			final List<JsonNode> extractions = new ArrayList<>();
+			documentIds.forEach(documentId -> {
+				final DocumentAsset document = documentProxy.getAsset(documentId).getBody();
+				final JsonNode documentExtractions = mapper.convertValue(document.getMetadata().get("attributes"), JsonNode.class);
+				final List<JsonNode> documentExtractionsAsList = mapper.convertValue(documentExtractions, new TypeReference<List<JsonNode>>() {});
+				extractions.addAll(documentExtractionsAsList);
+			});
+
+			model.setMetadata(model.getMetadata().setAttributes(extractions));
```

Comment:
You should be able to just do
`model.getMetadata().setAttributes(extractions);`

Reply:
And just to be clear... this will replace the list of attributes.  If you're looking to add to a list that may or may not be there it would be something like

```
if(model.getMetadata().getAttributes() == null)
    model.getMetadata().setAttributes(new ArrayList<>());
model.getMetadata().getAttributes().addAll(extractions)
```

Reply:
Interesting that an `getXXX().addAll()` will append, kind of becoming a `setXXX()`.

Reply:
I just find it weird, doesn't kind of negate the purpose of a setter then?